libraries/Microsoft.Bot.Builder.Dialogs.Declarative/ComponentRegistration.cs

```diff
@@ -1,27 +1,74 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
+﻿using System;
 using System.Collections.Generic;
-using Microsoft.Bot.Builder.Dialogs.Debugging;
-using Microsoft.Bot.Builder.Dialogs.Declarative.Resolvers;
-using Newtonsoft.Json;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Bot.Expressions;
 
 namespace Microsoft.Bot.Builder.Dialogs.Declarative
 {
     public class ComponentRegistration
     {
-        public ComponentRegistration()
+        /// <summary>
+        /// Gets component registrations.
+        /// </summary>
+        /// <value>
+        /// Component registrations object which implement IComponentRegistration or derived interface.
+        /// </value>
+        public static Lazy<List<ComponentRegistration>> Registrations { get; private set; } = new Lazy<List<ComponentRegistration>>(() =>
         {
-        }
+            void LoadReferencedAssembly(Assembly assembly)
```

Comment:
Reflection, dynamic assembly loading and recursion all in 16 lines of code. What could go wrong?